### PR TITLE
Add docker compose for the production and development server

### PR DESCRIPTION
**Pull request by vfurmane**

### CODE REVIEW

**Review by savosaicic — APPROVED** [review]

**Review by deblokc — APPROVED** [review]

**Review by soleraeus — APPROVED** [review]
Will have to switch to an nginx proxy on the long term (waiting for structure of nest api, especially regarding websockets, before implementing it). Added .dockerignore file and tweaked a couple commands before approving

File: docker-compose.yml [hunk 1]
```diff
@@ -5,6 +5,8 @@ services:
     build:
       context: .
       dockerfile: ./apps/web/Dockerfile
+    container_name: web
+    image: web:transcendence
```
Comment (vfurmane) [inline docker-compose.yml]: Why is the name of the image `web` and the tag `transcendence`? Isn't it supposed to be the name of the project and the version (e.g. ft_transcendence:latest)?

File: .dockerignore [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+**/Dockerfile
+.git
+.gitignore
+**/npm-debug.log
+**/.coverage
+**/.coverage.*
+**/.env
+**/.*.swp
+**/.vscode
+docker-compose.dev.yml
```
Comment (vfurmane) [inline .dockerignore:14]: Add ` docker-compose.yml` too

**Comment by soleraeus** [comment]
This is the name of the image of this specific container not of the
project, so it should be the name of the container with a tag.
Transcendence is the name of the whole project (which should be in .env btw
as
COMPOSE_PROJECT_NAME=ft_transcendence
)
Using latest tag is bad practice (some reading
https://vsupalov.com/docker-latest-tag/). For instance if you have another
project running another container named web with no tag, which is gonna be
autotagged latest, it's going to be a mess

**Review by savosaicic — APPROVED** [review]

**Review by soleraeus — APPROVED** [review]

**Review by maxenceeudier — APPROVED** [review]

**Pull request merged**